### PR TITLE
[WIP] Cleanup requirements more

### PR DESCRIPTION
**Pull request by mikemcquaid**
Following on from #38815. Still need to alias the requirements so they work with the old names.

### CODE REVIEW

**Comment by xu-cheng** [comment]
A little remind that `JavaDependency` is used inside `audit` [here](https://github.com/Homebrew/homebrew/blob/master/Library/Homebrew/cmd/audit.rb#L210).

**Comment by mikemcquaid** [comment]
Thanks @xu-cheng!

File: Library/Homebrew/compat/requirements.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+AprDependency            ||= AprRequirement
+FortranDependency        ||= FortranRequirement
+JavaDependency           ||= JavaRequirement
+JavaRequirement          ||= JavaRequirement
+LanguageModuleDependency ||= LanguageModuleRequirement
+MPIDependency            ||= MPIRequirement
+OsxfuseDependency        ||= OsxfuseRequirement
+PythonDependency         ||= PythonRequirement
+TuntapDependency         ||= TuntapRequirement
+X11Dependency            ||= X11Requirement
```
Comment (mikemcquaid) [inline Library/Homebrew/compat/requirements.rb]: @Homebrew/owners You can probably see what I'm trying to do here but I'm struggling with the syntax. Can anyone help?

Comment (xu-cheng) [inline Library/Homebrew/compat/requirements.rb]: You did everything right. The only thing you forget is to add `ConflictsWithBinaryOsxfuse ||= ConflictsWithBinaryOsxfuseRequirement`.

Comment (xu-cheng) [inline Library/Homebrew/compat/requirements.rb]: Also we can use this chance to rename it to something shorter.

Comment (mikemcquaid) [inline Library/Homebrew/compat/requirements.rb]: Thanks @xu-cheng! Any suggestions?

Comment (xu-cheng) [inline Library/Homebrew/compat/requirements.rb]: How about `NonBinaryOsxfuseRequirement` ? `ConflictsWithBinaryOsxfuseRequirement` just doesn't sound right.

**Comment by mikemcquaid** [comment]
@xu-cheng (or other @Homebrew/owners) Any idea about this: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/27914/version=mountain_lion/console ?

**Comment by xu-cheng** [comment]
I cannot reproduce this locally.

```
$ brew readalll

$ brew-1.8 readall # <- ruby 1.8.7-p35

```

**Comment by mikemcquaid** [comment]
@jacknagel Any ideas?

**Comment by jacknagel** [comment]
Weird. I'll look at it locally later.

**Comment by jacknagel** [comment]
It passed. *shrug*

**Pull request closed without merging**